src/views/Transitioner.js

```diff
@@ -235,6 +256,12 @@ class Transitioner extends React.Component<*, Props, State> {
     this._transitionProps = buildTransitionProps(this.props, nextState);
 
     this.setState(nextState, () => {
+      if (this._queuedTransition) {
+        this._startTransition(this._queuedTransition.nextProps, this._queuedTransition.nextScenes);
+        this._queuedTransition = null;
+      } else {
+        this._isTransitionRunning = false;
+      }
       this.props.onTransitionEnd && this.props.onTransitionEnd(
```

Comment:
I think this call to `onTransitionEnd` should be moved up to the start of this block. That way it is clear that it will be called before the `onTransitionStart` of the next transition